python/mlx/nn/layers/positional_encoding.py

```diff
@@ -75,31 +90,43 @@ def __call__(self, x, offset: int = 0):
         shape = x.shape
         x = mx.reshape(x, (-1, shape[-2], shape[-1]))
         N = x.shape[1] + offset
-        costheta, sintheta = RoPE.create_cos_sin_theta(
-            N, self.dims, offset=offset, base=self.base, scale=self.scale, dtype=x.dtype
-        )
+        if N > self.max_position:
```

Comment:
I am not sure if we need to cover the scenario where the sequence length is greater than the maximum position here. This could only happen when user passes the wrong maximum position during initialization. Since we are in the relatively early stage of mlx framework, maybe we should just throw an error here when the sequence length exceeds the maximum position so that people can fix their code.